cpu/cc2538/dev/spi.c

```diff
@@ -41,102 +41,337 @@
 #include "dev/spi.h"
 #include "dev/ssi.h"
 #include "dev/gpio.h"
+/*---------------------------------------------------------------------------*/
+/* Map the old port / pin names to the new ones for compatibility reasons */
+#if (SPI_DEFAULT_INSTANCE == 0)
+#ifdef SPI_CLK_PORT
+#define SPI0_CLK_PORT         SPI_CLK_PORT
+#endif
+#ifdef SPI_CLK_PIN
+#define SPI0_CLK_PIN          SPI_CLK_PIN
+#endif
+#ifdef SPI_MOSI_PORT
+#define SPI0_TX_PORT          SPI_MOSI_PORT
+#endif
+#ifdef SPI_MOSI_PIN
+#define SPI0_TX_PIN           SPI_MOSI_PIN
+#endif
+#ifdef SPI_MISO_PORT
+#define SPI0_RX_PORT          SPI_MISO_PORT
+#endif
+#ifdef SPI_MISO_PIN
+#define SPI0_RX_PIN           SPI_MISO_PIN
+#endif
+#elif (SPI_DEFAULT_INSTANCE == 1)
+#ifdef SPI_CLK_PORT
+#define SPI1_CLK_PORT         SPI_CLK_PORT
+#endif
+#ifdef SPI_CLK_PIN
+#define SPI1_CLK_PIN          SPI_CLK_PIN
+#endif
+#ifdef SPI_MOSI_PORT
+#define SPI1_TX_PORT          SPI_MOSI_PORT
+#endif
+#ifdef SPI_MOSI_PIN
+#define SPI1_TX_PIN           SPI_MOSI_PIN
+#endif
+#ifdef SPI_MISO_PORT
+#define SPI1_RX_PORT          SPI_MISO_PORT
+#endif
+#ifdef SPI_MISO_PIN
+#define SPI1_RX_PIN           SPI_MISO_PIN
+#endif
+#else
+#error "Invalid SPI instance. Valid values are 0 or 1"
+#endif
+/*---------------------------------------------------------------------------*/
+/* Check port / pin settings for SPI0 and provide default values for spi_cfg */
+#ifndef SPI0_CLK_PORT
+#define SPI0_CLK_PORT         (-1)
+#endif
+#ifndef SPI0_CLK_PIN
+#define SPI0_CLK_PIN          (-1)
+#endif
+#if SPI0_CLK_PORT >= 0 && SPI0_CLK_PIN < 0 || \
+  SPI0_CLK_PORT < 0 && SPI0_CLK_PIN >= 0
+#error "Both SPI0_CLK_PORT and SPI0_CLK_PIN must be valid or invalid"
+#endif
 
-#define SPI_CLK_PORT_BASE        GPIO_PORT_TO_BASE(SPI_CLK_PORT)
-#define SPI_CLK_PIN_MASK         GPIO_PIN_MASK(SPI_CLK_PIN)
-#define SPI_MOSI_PORT_BASE       GPIO_PORT_TO_BASE(SPI_MOSI_PORT)
-#define SPI_MOSI_PIN_MASK        GPIO_PIN_MASK(SPI_MOSI_PIN)
-#define SPI_MISO_PORT_BASE       GPIO_PORT_TO_BASE(SPI_MISO_PORT)
-#define SPI_MISO_PIN_MASK        GPIO_PIN_MASK(SPI_MISO_PIN)
+#ifndef SPI0_TX_PORT
+#define SPI0_TX_PORT          (-1)
+#endif
+#ifndef SPI0_TX_PIN
+#define SPI0_TX_PIN           (-1)
+#endif
+#if SPI0_TX_PORT >= 0 && SPI0_TX_PIN < 0 || \
+  SPI0_TX_PORT < 0 && SPI0_TX_PIN >= 0
+#error "Both SPI0_TX_PORT and SPI0_TX_PIN must be valid or invalid"
+#endif
```

Comment:
> Hi, what about SPI slave mode which might come at some day? RX / TX is unambiguous even in this case! And even in the users guide + data sheet the pins are name Tx and Rx...

OK, that makes sense. If part of this code is reused one day for slave SPI, using TX / RX would indeed avoid switching the pad names in the initialization. They use TX / RX in the user's guide because this SSI IP is not SPI-only. So keeping your TX / RX naming is justified.